tribe-ext-community-currency-symbol/index.php

```diff
@@ -0,0 +1,415 @@
+
+		$fields = array();
+
+		$fields['tribe_ext_ce_cost_currency_title'] = array(
+			'type' => 'html',
+			'html' => '<h3>' . esc_html__( 'Event Cost Currency Extension', 'tribe-extension' ) . '</h3>',
+		);
+
+		// List of allowed symbols
+		$fields['tribe_ext_ce_cost_currency_symbols'] = array(
+			'type'            => 'text',
+			'label'           => esc_html__( 'Allowed Symbols', 'tribe-extension' ),
+			'tooltip'         => esc_html__( 'Pipe-separated list of allowed currency symbols. (Once this option is set and saved, you will be able to pick the Default Currency Symbol for new events.) The drop-down will display these in the order you enter them. Example 1: $|€|£|Fr|¥|CNY|C$|AU$. Example 2: USD|EUR|GBP|JPY|CNY|CAD|AUD', 'tribe-extension' ),
+			'validation_type' => 'textarea',
+		);
+
+		// only allow modifying after the List exists (to be able to pick from)
+		if ( tribe_get_option( 'tribe_ext_ce_cost_currency_symbols' ) ) {
+			$fields['tribe_ext_ce_cost_currency_symbol_default'] = array(
+				'type'            => 'dropdown',
+				'label'           => esc_html__( 'Default Symbol', 'tribe-extension' ),
+				'tooltip'         => esc_html__( 'The default currency symbol for new events (does not apply when editing existing events). Leave blank to not set a default.', 'tribe-extension' ),
+				'options'         => $this->symbol_list_array( true ),
+				'validation_type' => 'options',
+			);
+		}
+
+		// List of allowed symbols
+		$fields['tribe_ext_ce_cost_currency_symbol_position_disabled'] = array(
+			'type'            => 'checkbox_bool',
+			'label'           => esc_html__( 'Disallow Position Selector', 'tribe-extension' ),
+			'tooltip'         => esc_html__( 'If checked, event creators will not be able to change the currency position (before or after the cost) and your site default from the General settings tab will be used.', 'tribe-extension' ),
+			'default'         => false,
+			'validation_type' => 'boolean',
+		);
+
+		$setting_helper->add_fields(
+			$fields,
+			'community',
+			'single_geography_mode' // At the bottom of the "Form Defaults" section
+		);
+	}
+
+	/**
+	 * Get list of allowed symbols as an array.
+	 *
+	 * @param bool $prepend_empty
+	 *
+	 * @return array
+	 */
+	public function symbol_list_array( $prepend_empty = false ) {
+		$setting = tribe_get_option( 'tribe_ext_ce_cost_currency_symbols' );
+
+		if ( '' === (string) $setting ) {
+			return array();
+		}
+
+		$settings = explode( '|', $setting );
+
+		/**
+		 * Filters the array of currency symbols passed to the drop-down.
+		 *
+		 * @param array $settings The original array.
+		 */
+		$settings = (array) apply_filters( 'tribe_ext_ce_cost_currency_symbol_list_array', $settings );
+
+		$symbols = array();
+
+		foreach ( $settings as $key => $value ) {
+			$value = esc_attr( sanitize_text_field( $value ) );
+			if ( '' !== $value ) {
+				$symbols[ $value ] = $value;
+			}
+		}
+
+		(bool) $prepend_empty;
```

Comment:
This won't actually do anything. If we want to capture the result of the cast we need:

```php
$prepend_empty = (bool) $prepend_empty;
```

Reply:
ooh, good catch!